recommend/imageHandler.go

```diff
@@ -477,37 +478,76 @@ func getImageDetails(img ImageInfo) error {
 	// step 3: getImageInfo
 	img.getImageInfo()
 
+	if len(img.RepoTags) == 0 {
+		img.RepoTags = append(img.RepoTags, img.Name)
+	}
 	// step 4: get policy from image info
 	img.getPolicyFromImageInfo()
 
 	return nil
 }
 
-func imageHandler(namespace, deployment string, labels LabelMap, imageName string, config string) error {
-	dockerConfigPath = config
-	log.WithFields(log.Fields{
-		"image": imageName,
-	}).Info("pulling image")
+func imageHandler(namespace, deployment string, labels LabelMap, imageName string, c *k8s.Client) error {
+	dockerConfigPath = options.Config
 	img := ImageInfo{
 		Name:       imageName,
 		Namespace:  namespace,
 		Deployment: deployment,
 		Labels:     labels,
 	}
+	policiesToBeRecommended := options.Policy
+	reportStarted := false
+
+	for _, policyToBeRecommended := range policiesToBeRecommended {
+		switch policyToBeRecommended {
+		case KyvernoPolicy:
+			if len(img.RepoTags) == 0 {
+				img.RepoTags = append(img.RepoTags, img.Name)
+			}
+			if !reportStarted {
+				if err := ReportStart(&img); err != nil {
+					log.WithError(err).Error("report start failed")
+					return err
+				}
+				reportStarted = true
+			}
```

Comment:
It's a bit confusing, what happens if KyvernoPolicy is before KubeArmor in the array? will it do 2 ReportStart?

Reply:
Changed the approach.